src/meta/src/manager/catalog/mod.rs

```diff
@@ -795,6 +796,18 @@ where
                 .map(|(index_id, index)| (*index_id, index.index_table_id))
                 .unzip();
 
+            let mut index_internal_table_ids = Vec::with_capacity(index_table_ids.len());
+            for index_table_id in &index_table_ids {
+                let internal_table_ids = fragment_manager
+                    .select_table_fragments_by_table_id(&(index_table_id.into()))
+                    .await?
+                    .internal_table_ids();
+
+                // Only 1 should be used by table scan.
+                assert_eq!(internal_table_ids.len(), 1);
```

Comment:
This seems to be a breaking change. For indexes created prior to this PR, I guess this assertion will fail.

Reply:
Is tracking it in release notes of this PR sufficient?

Reply:
I guess not. We have to handle old case too.

Reply:
Assertion `<=` should be sufficient here?

Reply:
Assertion `<=` should be enough here?